build.sbt

```diff
@@ -250,6 +253,17 @@ lazy val `kyo-direct` =
         .nativeSettings(`native-settings`)
         .jsSettings(`js-settings`)
 
+lazy val `kyo-stm` =
```

Comment:
I've moved the implementation to a separate module since we'll likely have multiple STM-based implementations like `TPromise`, `TChunk`, `TQueue`, etc. It'll also help reduce compilation times of `kyo-core`.